mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/migration/MailboxPathV2MigrationTest.java

```diff
@@ -0,0 +1,133 @@
+import org.apache.james.mailbox.cassandra.mail.CassandraUserMailboxRightsDAO;
+import org.apache.james.mailbox.cassandra.modules.CassandraAclModule;
+import org.apache.james.mailbox.cassandra.modules.CassandraMailboxModule;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.store.mail.model.impl.SimpleMailbox;
+import org.assertj.core.api.SoftAssertions;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+public class MailboxPathV2MigrationTest {
+
+    public static final MailboxPath MAILBOX_PATH_1 = MailboxPath.forUser("bob", "Important");
+    public static final int UID_VALIDITY_1 = 452;
+    public static final SimpleMailbox MAILBOX_1 = new SimpleMailbox(MAILBOX_PATH_1, UID_VALIDITY_1);
+    public static final CassandraId MAILBOX_ID_1 = CassandraId.timeBased();
+
+    static {
+        MAILBOX_1.setMailboxId(MAILBOX_ID_1);
+    }
+
+    @ClassRule
+    public static DockerCassandraRule cassandraServer = new DockerCassandraRule();
+
+    private CassandraMailboxPathDAOImpl daoV1;
+    private CassandraMailboxPathV2DAO daoV2;
+    private CassandraMailboxMapper mailboxMapper;
+    private CassandraMailboxDAO mailboxDAO;
+
+    @Before
+    public void setUp() {
+        CassandraCluster cassandra = CassandraCluster.create(
+            new CassandraModuleComposite(
+                new CassandraMailboxModule(),
+                new CassandraAclModule()),
+            cassandraServer.getIp(),
+            cassandraServer.getBindingPort());
+
+        daoV1 = new CassandraMailboxPathDAOImpl(
+            cassandra.getConf(),
+            cassandra.getTypesProvider(),
+            CassandraUtils.WITH_DEFAULT_CONFIGURATION);
+        daoV2 = new CassandraMailboxPathV2DAO(
+            cassandra.getConf(),
+            CassandraUtils.WITH_DEFAULT_CONFIGURATION);
+
+        CassandraUserMailboxRightsDAO userMailboxRightsDAO = new CassandraUserMailboxRightsDAO(cassandra.getConf(), CassandraUtils.WITH_DEFAULT_CONFIGURATION);
+        mailboxDAO = new CassandraMailboxDAO(cassandra.getConf(), cassandra.getTypesProvider());
+        mailboxMapper = new CassandraMailboxMapper(
+            mailboxDAO,
+            daoV1,
+            daoV2,
+            userMailboxRightsDAO,
+            new CassandraACLMapper(cassandra.getConf(), userMailboxRightsDAO, CassandraConfiguration.DEFAULT_CONFIGURATION));
+    }
+
+    @Test
+    public void newValuesShouldBeSavedInMostRecentDAO() throws Exception {
+        mailboxMapper.save(MAILBOX_1);
+
+        assertThat(daoV2.retrieveId(MAILBOX_PATH_1).join())
+            .contains(new CassandraIdAndPath(MAILBOX_ID_1, MAILBOX_PATH_1));
+    }
+
+    @Test
+    public void newValuesShouldNotBeSavedInOldDAO() throws Exception {
+        mailboxMapper.save(MAILBOX_1);
+
+        assertThat(daoV1.retrieveId(MAILBOX_PATH_1).join())
+            .isEmpty();
+    }
+
+    @Test
+    public void readingOldValuesShouldNotMigrateThem() throws Exception {
```

Comment:
why didn't you choose that strategy ?

Reply:
Time of implementation.

If you want I can code it before the week-end :trollface:

Reply:
ok